### DISCUSSION TITLE
为什么 Linux 默认页大小是 4KB - 面向信仰编程 · /whys-the-design-linux-default-page

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#618] [Announcements]
https://draveness.me/whys-the-design-linux-default-page/

**Comment by pengpaige** [comment]
手动点赞

**Comment by wolf-joe** [comment]
补充两点从《现代操作系统》里看到的：
* HDD的寻道时间比传输时间长很多。内存、HDD之间的传输一般是一次一页，所以传2个8k的页（2\*12ms）比传4个4k的页（4\*10ms）性能更好。
* 某些系统切换进程时需要将新进程的页表装入寄存器，页表更小的大页面性能更好。（不过在x86上切换进程时好像只需要修改页表指针寄存器，所以没区别？）

**Comment by yuanlinZhao** [comment]
过大的页面大小会浪费内存空间，造成内存碎片，降低内存的利用率，但是可以较少进程中的页表项以及 TLB 的寻址时间；

应该是减少进程中的页表项吧

---

*2021-01-12 UPDATES: 已修复*

**Comment by yuanlinZhao** [comment]
顺便在此处咨询下 为什么 MongoDB 使用 B 树 打不开了，期望修复可以阅读~

**Comment by draveness** [comment]
> 过大的页面大小会浪费内存空间，造成内存碎片，降低内存的利用率，但是可以较少进程中的页表项以及 TLB 的寻址时间；
> 
> 应该是减少进程中的页表项吧

已修复


> 顺便在此处咨询下 为什么 MongoDB 使用 B 树 打不开了，期望修复可以阅读~

这篇文章有问题，已经删掉了，之后重写

**Comment by cdpath** [comment]
"增加寻址时 TLB（Translation lookaside buffer）的查找速度"

---

s/查找速度/查找时间

**Comment by wxainn** [comment]
看你的文章真是一种享受，已订阅

**Comment by nonbutAworker** [comment]
``在如上图所示的四层页表结构中，操作系统会使用最低的 12 位作为页面的偏移量，剩下的 36 位会分四组分别表示当前层级在上一层中的索引，所有的虚拟地址都可以用上述的多层页表查找到对应的物理地址``
这个地方我看文章里说的是64位系统，去掉12位用来表示页表偏移量，剩下的不应该是52位吗，为什么这里只剩下36位了，36+12也不等于64啊

**Reply by yangxuhui** [reply]
目前的CPU只使用48位虚拟地址空间，因为48位构成了256TB的地址空间，这已经远超绝大多数系统需要的地址空间。可以使用`cat /proc/cpuinfo`查看CPU信息，结果中有一行 `address sizes : 46 bits physical, 48 bits virtual`

**Comment by gsonkeno** [comment]
搜索
